src/util/Generator.h

```diff
@@ -5,22 +5,22 @@
 #ifndef CPPCORO_GENERATOR_HPP_INCLUDED
 #define CPPCORO_GENERATOR_HPP_INCLUDED
 
-// coroutines are still experimental in clang, adapt the appropriate
-// namespaces.
-#ifdef __clang__
+#include <exception>
+#include <functional>
+#include <iterator>
+#include <type_traits>
+#include <utility>
+
+// coroutines are still experimental in clang libcpp,
+// adapt the appropriate namespaces.
+#ifdef _LIBCPP_VERSION
```

Comment:
`_LIBCPP_VERSION` will be defined by one of the "standard headers", that's why it has to be moved below other headers.